cmd/opampsupervisor/supervisor/commander/commander.go

```diff
@@ -102,17 +104,19 @@ func (c *Commander) watch() {
 
 	c.running.Store(0)
 	close(c.doneCh)
+	c.exitCh <- struct{}{}
```

Comment:
This will block if there are no listeners right now. We should buffer this with length `1`, then clear it on startup.